pkg/detectors/tableau/tableau.go

```diff
@@ -0,0 +1,257 @@
+package tableau
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"maps"
+	"net/http"
+	"strings"
+
+	regexp "github.com/wasilibs/go-re2"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct {
+	detectors.DefaultMultiPartCredentialProvider
+	detectors.EndpointSetter
+	client *http.Client
+}
+
+// Ensure the Scanner satisfies the interface at compile time.
+var _ detectors.Detector = (*Scanner)(nil)
+var _ detectors.EndpointCustomizer = (*Scanner)(nil)
+
+func (Scanner) CloudEndpoint() string { return "" }
+
+var (
+	defaultClient = common.SaneHttpClient()
+
+	// Simplified token name pattern using PrefixRegex
+	tokenNamePat = regexp.MustCompile(detectors.PrefixRegex([]string{"token_name", "pat_name", "tableau_pat_name", "tableau_token", "pat", "token", "name"}) + `\s*["']?([a-zA-Z0-9_-]{3,50})["']?`)
```

Comment:
```
([a-zA-Z][a-zA-Z0-9_-]{2,50})
```
This regex ensures that the first character is always a letter, so there's no need for additional validation checks later.

Reply:
Including `\s*["']?` and `["']?` is unnecessary.